### PR TITLE
Add more test cases for sum-buddy

### PR DESCRIPTION
**Pull request by zoeduan**
Add more test cases for sum-buddy:
- test_mapper.py: This test file contains unit tests for the mapper module of the sum-buddy project. 
- test_hasher.py: This test file includes unit tests for the hasher module of the sum-buddy project. 
- test_getChecksums.py: This test file comprises unit tests for the get_checksums function in the sum-buddy project. 
- test_filter.py: This test file contains unit tests for the filter module of the sum-buddy project. 

Add tests #4

### CODE REVIEW

**Comment by thompsonmj** [comment]
Tests fail for the SHAKE algorithms for me. Seems it's due to a [length parameter being required ](https://docs.python.org/3/library/hashlib.html#shake-variable-length-digests). The `checksum_file` method will need updated accordingly. 
Though oddly, what would be intuitive to set as a digest byte length for e.g. `shake_128` of 16, the 128 refers to the maximum security strength of the algorithm which requires a length setting of at least 32 (similarly for `shake_256`, [OpenSSL docs](https://www.openssl.org/docs/man3.0/man1/openssl-dgst.html)).
Even more fun, it looks like the BLAKES can be [variable length](https://docs.python.org/3/library/hashlib.html#creating-hash-objects) as well, though that setting is not needed to produce an output.

I can add these nuances to `checksum_file`.

**Comment by thompsonmj** [comment]
All tests pass for me now. Though I think it would be good to test error handling (passing length parameter to a fixed-length algorithm) and some other edge cases, like trying to checksum a file that doesn't exist.

**Comment by egrace479** [comment]
> All tests pass for me now. Though I think it would be good to test error handling (passing length parameter to a fixed-length algorithm) and some other edge cases, like trying to checksum a file that doesn't exist.

All passing for me as well. 

I agree; @zoeduan, do you think you could add tests for those two cases?

**Comment by zoeduan** [comment]
Thank you! @thompsonmj @egrace479 I will add tests for those two cases.

**Comment by zoeduan** [comment]
> Thank you! @thompsonmj @egrace479 I will add tests for those two cases.

I added test error handling (passing length parameter to a fixed-length algorithm) and 
edge case when trying to checksum a file that doesn't exist.

**Review by thompsonmj — APPROVED** [review]
LGTM

File: tests/test_hasher.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    def test_checksum_file_sha512(self):
+        self._test_checksum_file_with_algorithm("sha512")
+
+    def test_checksum_file_blake2b(self):
+        self._test_checksum_file_with_algorithm("blake2b")
+
+    def test_checksum_file_blake2s(self):
+        self._test_checksum_file_with_algorithm("blake2s")
+
+    def test_checksum_file_sha3_224(self):
```
Comment (egrace479) [inline tests/test_hasher.py:77]:
```suggestion
    def test_checksum_file_blake2b_len(self):
        self._test_checksum_file_with_algorithm("blake2b", length = 64)

    def test_checksum_file_blake2s_len(self):
        self._test_checksum_file_with_algorithm("blake2s", length = 32)

    def test_checksum_file_sha3_224(self):
```

Since the blakes also have the option for a length parameter, we should probably test them passed with length as well.
